src/dotnet/GeneratorAgent/tests/Azure.Tools.GeneratorAgent.Tests/ProcessExecutorTests.cs

```diff
@@ -0,0 +1,302 @@
+using Azure.Tools.GeneratorAgent;
+using Microsoft.Extensions.Logging;
+using Moq;
+using NUnit.Framework;
+using System.ComponentModel;
+using System.Diagnostics;
+
+namespace Azure.Tools.GeneratorAgent.Tests
+{
+    [TestFixture]
+    public class ProcessExecutorTests
+    {
+        private const string ValidCommand = "cmd.exe";
+        private const string EchoArguments = "/c echo test message";
+        private const string FailArguments = "/c exit 1";
+        private const string ValidWorkingDirectory = @"C:\Windows\System32";
+        private const string ExpectedOutput = "test message";
+        private const int TimeoutMilliseconds = 5000;
+
+        private Mock<ILogger<ProcessExecutor>> mockLogger = null!;
+        private ProcessExecutor executor = null!;
+        private CancellationTokenSource cancellationTokenSource = null!;
+
+        [SetUp]
+        public void SetUp()
+        {
+            mockLogger = new Mock<ILogger<ProcessExecutor>>();
```

Comment:
Do we need class-level state?   Better to keep tests fully isolated and create/destroy everything in the test itself.